src/plugins/controls/public/options_list/components/options_list_popover_suggestions.tsx

```diff
@@ -49,130 +43,99 @@ export const OptionsListPopoverSuggestions = ({
-          </p>
-        </div>
-      </div>
-    );
-  }
+  const suggestions = useMemo(() => {
+    return showOnlySelected ? selectedOptions : Object.keys(availableOptions ?? {});
+  }, [availableOptions, selectedOptions, showOnlySelected]);
+
+  const existsSelectableOption = useMemo<EuiSelectableOption | undefined>(() => {
+    if (hideExists || (!existsSelected && (showOnlySelected || suggestions?.length === 0))) return;
+
+    return {
+      key: 'exists-option',
+      checked: existsSelected ? 'on' : undefined,
+      label: OptionsListStrings.controlAndPopover.getExists(),
+      className: 'optionsList__existsFilter',
+      'data-test-subj': 'optionsList-control-selection-exists',
+    };
+  }, [suggestions, existsSelected, showOnlySelected, hideExists]);
+
+  const suggestionsSelectableOptions = useMemo<EuiSelectableOption[]>(() => {
+    const options: EuiSelectableOption[] = (suggestions ?? []).map((key) => {
+      return {
+        key,
+        label: key,
+        checked: selectedOptionsSet?.has(key) ? 'on' : undefined,
+        'data-test-subj': `optionsList-control-selection-${key}`,
+        className:
+          showOnlySelected && invalidSelectionsSet.has(key)
+            ? 'optionsList__selectionInvalid'
+            : 'optionsList__validSuggestion',
+        append:
+          !showOnlySelected && availableOptions?.[key] ? (
+            <OptionsListPopoverSuggestionBadge documentCount={availableOptions[key].doc_count} />
+          ) : undefined,
+      };
+    });
+    return existsSelectableOption ? [existsSelectableOption, ...options] : options;
+  }, [
+    suggestions,
+    availableOptions,
+    showOnlySelected,
+    selectedOptionsSet,
+    invalidSelectionsSet,
+    existsSelectableOption,
+  ]);
+
+  const [selectableOptions, setSelectableOptions] = useState<EuiSelectableOption[]>([]); // will be set in following useEffect
+  useEffect(() => {
+    /* This useEffect makes suggestionsSelectableOptions responsive to search, show only selected, and clear selections */
+    setSelectableOptions(suggestionsSelectableOptions);
+  }, [suggestionsSelectableOptions]);
 
   return (
-    <>
-      {!hideExists && !(showOnlySelected && !existsSelected) && (
-        <EuiFilterSelectItem
-          data-test-subj={`optionsList-control-selection-exists`}
-          checked={existsSelected ? 'on' : undefined}
-          key={'exists-option'}
-          onClick={() => {
-            dispatch(selectExists(!Boolean(existsSelected)));
-          }}
-          className="optionsList__existsFilter"
-        >
-          {OptionsListStrings.controlAndPopover.getExists()}
-        </EuiFilterSelectItem>
+    <EuiSelectable
+      isLoading={isLoading}
+      loadingMessage={
+        <span data-test-subj="optionsList-control-popover-loading">
+          <EuiLoadingSpinner size="m" />
+          <EuiSpacer size="xs" />
+          {OptionsListStrings.popover.getLoadingMessage()}
+        </span>
```

Comment:
I had to re-create the default loading message for the sake of functional testing, so that I could give the component a `data-test-subj`. It looks like this:

![Jan-12-2023 12-23-56](https://user-images.githubusercontent.com/8698078/212161832-544595e0-5fc3-4ba4-95ca-971b69950fcc.gif)